db/blob/blob_source.cc

```diff
@@ -134,7 +134,15 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
     Slice key = cache_key.AsSlice();
     s = GetBlobFromCache(key, &blob_entry);
     if (s.ok() && blob_entry.GetValue()) {
-      value->PinSelf(*blob_entry.GetValue());
+      // To avoid copying the cached blob into the buffer provided by the
+      // application, we can simply transfer ownership of the cache handle to
+      // the target PinnableSlice. This has the potential to save a lot of
+      // CPU, especially with large blob values.
+      value->data_ = blob_entry.GetValue()->data();
+      value->size_ = blob_entry.GetValue()->size();
+      value->Pin();
+      assert(value->IsPinned());
+      blob_entry.TransferTo(value);
```

Comment:
Not an expert. I wonder whether this can be wrapped in a specialized template member function of `CachableEntry<std::string>::TransferTo(PinnableSlice*)`.

Reply:
Yeah, sure. I can change that.

Reply:
We probably can only use `CachableEntry<std::string>::TransferTo(Cleanable*)`.